deploy.py

```diff
@@ -1,124 +1,152 @@
+                     'ParameterValue': get_bucket_name(config),
+                     'UsePreviousValue': False},
+                    {'ParameterKey': 'KeyName',
+                     'ParameterValue': config['ssh_key_name'],
+                     'UsePreviousValue': False},
+                    {'ParameterKey': 'Tag',
+                     'ParameterValue': config['tag'],
+                     'UsePreviousValue': False}],
+        Capabilities=['CAPABILITY_NAMED_IAM'],
     )
-    print("deployed stack!")
+    print("Uploaded deployment spec. CloudFormation will take it from here.")
+
+
+def upload_ssh_key(key_name):
+    """Upload the contents of the local SSH key into the secret."""
+    id = "{}.pem".format(key_name)
+    secrets.update_secret(SecretId=id, SecretString=open(id, 'r').read())
+    print("Uploaded SSH key.")
+
+
+def fail(msg):
+    """Print and quit."""
+    print(msg)
+    sys.exit(1)
+
 
 if __name__ == "__main__":
 
-    # The only argument required is the tag that makes this deployment and all its resources unique.
+    # The only argument required is the tag that makes this deployment and all
+    # its resources unique.
     parser = argparse.ArgumentParser()
-    parser.add_argument("--tag", "-t", required=False, help="tag to build, must be alphanumeric like \"prod\" or \"test\"")
+    parser.add_argument(
+        "--tag", "-t",
+        required=False,
+        default="test",
+        help="tag to build, must be alphanumeric like \"prod\" or \"test\"")
+
+    parser.add_argument(
+        "--project", "-p",
+        required=False,
+        default="umsi-easy-hub",
+        help="name of project, used in all AWS resources")
 
     args = parser.parse_args()
 
-    # Default tag is "test"
-    if args.tag is None:
-        tag = "test"
-    else:
-        tag = args.tag
+    # We plan to allow different names, but this project name is hard coded all
+    # over this project. We need to convert it to an input.
+    if args.project != "umsi-easy-hub":
+        fail("Using a different project name is currently unsupported.")
+
+    # This is a current limitation due to our naming scheme.
+    # An ELBV2::TargetGroup has a max name length of 32.
+    # Our template name is: ueh-${Tag}-target-group-https, leaving 9 for tag.
+    # The control node will pass, but the cluster will fail.
+    if len(args.tag) > 9:
+        fail("Due to limitations in AWS, the tag name must have a length of "
+             "at most 9. Your tag '{}' has length {}.".format(
+                 args.tag, len(args.tag)))
 
     # Generate basic config
     config = {}
-    config['tag'] = tag
-    config['project'] = project
-    config['account_id'] = boto3.client('sts').get_caller_identity().get('Account')
+    config['tag'] = args.tag
+    config['project'] = args.project
+    config['account_id'] = boto3.client(
+        'sts').get_caller_identity().get('Account')
     config['ssh_key_name'] = generate_ssh_key(config)
     print(config)
```

Comment:
We should eventually use proper logging here instead of print, but that's for another PR...